### DISCUSSION TITLE
how to handle sending/receiving big data with request-response protocol

### DISCUSSION THREAD

**Discussion by snowmead** [libp2p/rust-libp2p#5016] [General]
With a send request is executed to the request-response protocol, how can the receiver of the data process chunks of it at a time until the entirerity of the data has been sent from the sender? 

In the case where the sender transmits GBs of data, I want to reduce memory usage on the receiver's end by streaming chunks instead of loading everything into memory first and then responding to the channel that intiated the request.

Are there any examples of this, or any pointers on how I can achieve this?

**Comment by thomaseizinger** [comment]
I pointed at a few things in https://github.com/libp2p/rust-libp2p/issues/5018 :)

**Reply by snowmead** [reply]
Thank you!
